tests/src/Traits/InspectTestRequestTrait.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\Tests\EuropaSearchClient\Traits;
+
+use Psr\Http\Message\RequestInterface;
+
+trait InspectTestRequestTrait
+{
+    /**
+     * @var string
+     */
+    protected $boundary;
+
+    /**
+     * @param RequestInterface $request
+     */
+    protected function inspectTokenRequest(RequestInterface $request): void
+    {
+        $this->assertEquals('http://example.com/token', $request->getUri());
+        $this->assertSame('Basic YmF6OnF1eA==', $request->getHeaderLine('Authorization'));
+        $this->assertSame('application/x-www-form-urlencoded', $request->getHeaderLine('Content-Type'));
+        $this->assertSame('grant_type=client_credentials', $request->getBody()->getContents());
+    }
+
+    /**
+     * @param RequestInterface $request
+     */
+    protected function inspectAuthorizationHeaders(RequestInterface $request): void
+    {
+        $this->assertSame('Bearer JWT_TOKEN', $request->getHeaderLine('Authorization'));
+        $this->assertSame('JWT_TOKEN', $request->getHeaderLine('Authorization-propagation'));
+    }
+
+    /**
+     * @param RequestInterface $request
+     */
+    protected function inspectBoundary(RequestInterface $request): void
+    {
+        preg_match('/; boundary="([^"].*)"/', $request->getHeaderLine('Content-Type'), $found);
+        $this->boundary = $found[1];
```

Comment:
Using a property to store the boundary to run assertions across methods, it stinks. Why not extract the boundary and pass it to the methods that rely on it?

Reply:
done